### ISSUE TITLE
getaddrinfo may spin forever

### ISSUE THREAD

**Issue by greatitman** [signalwire/freeswitch#1466]
When freeswitch calls zoom, the call will get stuck occasionally, and normal SIP negotiation cannot be completed. GDB follows up the code when the card is dead. It shows that the program is stuck in the switch_ sockaddr_ info_ On the get function, go further into the GDB and find that it is finally stuck in getaddrinfo. The stack is as follows:
 
 - Version  1.10.4-dev]

``` 
Thread 24 (Thread 0x7f229cd7e700 (LWP 46536)):
#0  0x00007f22f225de1d in recvmsg () from /lib64/libc.so.6
#1  0x00007f22f2283c7d in make_request () from /lib64/libc.so.6
#2  0x00007f22f22841b4 in __check_pf () from /lib64/libc.so.6
#3  0x00007f22f2245ea1 in getaddrinfo () from /lib64/libc.so.6
#4  0x00007f22f5373f96 in call_resolver (sa=0x7f22747520b0, sa@entry=0x7f22601f3828, hostname=0x7f22601f37f0 "123.11.141.17", hostname@entry=0xca0c <Address 0xca0c out of bounds>, family=0, port=port@entry=21724, p=0x7f2260692b58, flags=0) at network_io/unix/sockaddr.c:337
#5  0x00007f22f5374542 in find_addresses (p=<optimized out>, flags=0, port=21724, family=<optimized out>, hostname=0xca0c <Address 0xca0c out of bounds>, sa=0x7f22601f3828) at network_io/unix/sockaddr.c:431
#6  apr_sockaddr_info_get (sa=sa@entry=0x7f22601f3828, hostname=hostname@entry=0xca0c <Address 0xca0c out of bounds>, family=family@entry=0, port=port@entry=21724, flags=flags@entry=0, p=<optimized out>) at network_io/unix/sockaddr.c:568
#7  0x00007f22f50119c8 in switch_sockaddr_info_get (sa=sa@entry=0x7f22601f3828, hostname=hostname@entry=0xca0c <Address 0xca0c out of bounds>, family=family@entry=0, port=port@entry=21724, flags=flags@entry=0, pool=<optimized out>) at src/switch_apr.c:834
#8  0x00007f22f50b9cb9 in switch_rtp_set_local_address (rtp_session=0x7f22601f37f0, host=0xca0c <Address 0xca0c out of bounds>, host@entry=0x7f22601f37f0 "123.11.141.17", port=port@entry=21724, err=0xca0c, err@entry=0x7f229cd7c578) at src/switch_rtp.c:2760
#9  0x00007f22f50bb143 in switch_rtp_new (rx_host=0x7f22601f37f0 "123.11.141.17", rx_port=<optimized out>, tx_host=0x7f2274802440 "221.123.139.199", tx_port=<optimized out>, payload=<optimized out>, samples_per_interval=<optimized out>, ms_per_packet=10000, flags=flags@entry=0x7f229cd7c5c0, timer_name=timer_name@entry=0x158a3f0 "soft", err=err@entry=0x7f229cd7c578, pool=0x7f2260692b58, bundle_internal_port=bundle_internal_port@entry=0, bundle_external_port=bundle_external_port@entry=0) at src/switch_rtp.c:4593
#10 0x00007f22f5069c5d in switch_core_media_activate_rtp (session=0x7f22601e7ad8) at src/switch_core_media.c:9184
#11 0x00007f22d7b79dde in sofia_media_activate_rtp (tech_pvt=tech_pvt@entry=0x7f22601f3318) at sofia_media.c:58
#12 0x00007f22d7b520e2 in sofia_handle_sip_i_state (de=0x7f22cc015ad0, tags=0x7f22bc0358e0, sip=0x0, sofia_private=<optimized out>, nh=0x7f227452fb80, profile=0x1588640, nua=0x7f22cc021f40, phrase=0x7f22bc035920 "ACK sent", status=200, session=0x7f22601e7ad8) at sofia.c:8502
#13 our_sofia_event_callback (event=nua_i_state, status=<optimized out>, phrase=0x7f22bc035920 "ACK sent", nua=0x7f22cc021f40, profile=0x1588640, nh=0x7f227452fb80, sofia_private=0x7f2274a77740, sip=0x0, de=de@entry=0x7f22cc015ad0, tags=0x7f22bc0358e0) at sofia.c:1830
#14 0x00007f22d7b52e9b in sofia_process_dispatch_event (dep=dep@entry=0x7f229cd7d250) at sofia.c:2270
#15 0x00007f22d7b17859 in sofia_receive_message (session=0x7f22601e7ad8, msg=0x7f229cd7d9c0) at mod_sofia.c:1331
#16 0x00007f22f50407e5 in switch_core_session_perform_receive_message (session=session@entry=0x7f22601e7ad8, message=message@entry=0x7f229cd7d9c0, file=file@entry=0x7f22f53d3925 "src/switch_ivr.c", func=func@entry=0x7f22f53d4930 <__func__.18705> "switch_ivr_parse_signal_data", line=line@entry=893) at src/switch_core_session.c:851
#17 0x00007f22f50fbc4c in switch_ivr_parse_signal_data (session=session@entry=0x7f22601e7ad8, all=all@entry=SWITCH_TRUE, only_session_thread=only_session_thread@entry=SWITCH_FALSE) at src/switch_ivr.c:893
#18 0x00007f22f50fbc9c in switch_ivr_parse_all_signal_data (session=session@entry=0x7f22601e7ad8) at src/switch_ivr.c:906
#19 0x00007f22f50fbcb7 in switch_ivr_parse_all_messages (session=session@entry=0x7f22601e7ad8) at src/switch_ivr.c:852
#20 0x00007f22f50fe52c in switch_ivr_parse_all_events (session=session@entry=0x7f22601e7ad8) at src/switch_ivr.c:925
#21 0x00007f22f5048254 in switch_core_session_run (session=0x7f22601e7ad8) at src/switch_core_state_machine.c:749
#22 0x00007f22f50424ce in switch_core_session_thread (thread=<optimized out>, obj=0x7f22601e7ad8) at src/switch_core_session.c:1732
#23 0x00007f22f503de13 in switch_core_session_thread_pool_worker (thread=0x7f22601f97b0, obj=<optimized out>) at src/switch_core_session.c:1795
#24 0x00007f22f537798c in dummy_worker (opaque=0x7f22601f97b0) at threadproc/unix/thread.c:151
#25 0x00007f22f2533dd5 in start_thread () from /lib64/libpthread.so.0
#26 0x00007f22f225cead in clone () from /lib64/libc.so.6
```

**Comment by greatitman** [comment]
Thread 27 (Thread 0x7f229e3b3700 (LWP 40282)):
#0  0x00007f22f2537965 in pthread_cond_wait@@GLIBC_2.3.2 () from /lib64/libpthread.so.0
#1  0x00007f22f53692fd in apr_queue_pop (queue=0x14b4ca8, data=0x7f229e3b2d40) at misc/apr_queue.c:276
#2  0x00007f22f5011ed5 in switch_queue_pop (queue=<optimized out>, data=<optimized out>) at src/switch_apr.c:1159
#3  0x00007f22f509f337 in chat_thread_run (thread=<optimized out>, obj=0x14b4ca8) at src/switch_loadable_module.c:946
#4  0x00007f22f537798c in dummy_worker (opaque=0x16a7700) at threadproc/unix/thread.c:151
#5  0x00007f22f2533dd5 in start_thread () from /lib64/libpthread.so.0
#6  0x00007f22f225cead in clone () from /lib64/libc.so.6
Thread 26 (Thread 0x7f229e376700 (LWP 40283)):
#0  0x00007f22f2537965 in pthread_cond_wait@@GLIBC_2.3.2 () from /lib64/libpthread.so.0
#1  0x00007f22f53692fd in apr_queue_pop (queue=0x16a7730, data=0x7f229e375d40) at misc/apr_queue.c:276
#2  0x00007f22f5011ed5 in switch_queue_pop (queue=<optimized out>, data=<optimized out>) at src/switch_apr.c:1159
#3  0x00007f22f509f337 in chat_thread_run (thread=<optimized out>, obj=0x16a7730) at src/switch_loadable_module.c:946
#4  0x00007f22f537798c in dummy_worker (opaque=0x16b39a0) at threadproc/unix/thread.c:151
#5  0x00007f22f2533dd5 in start_thread () from /lib64/libpthread.so.0
#6  0x00007f22f225cead in clone () from /lib64/libc.so.6
Thread 25 (Thread 0x7f229d5bc700 (LWP 202434)):
#0  0x00007f22f225de1d in recvmsg () from /lib64/libc.so.6
#1  0x00007f22f228177d in __netlink_request () from /lib64/libc.so.6
#2  0x00007f22f2281a2f in getifaddrs_internal () from /lib64/libc.so.6
#3  0x00007f22f2282760 in getifaddrs () from /lib64/libc.so.6
#4  0x00007f22f50a5366 in get_netmask (mask=0x7f229d5bb8d4, me=0x7f229d5bb850) at src/switch_utils.c:1685
#5  switch_find_local_ip (buf=buf@entry=0x7f229d5bb8e0 "10.50.128.17", len=len@entry=256, mask=mask@entry=0x7f229d5bb8d4, family=family@entry=2) at src/switch_utils.c:1950
#6  0x00007f22f504f091 in check_ip () at src/switch_core.c:144
#7  0x00007f22f504f569 in check_ip_callback (task=0x14a0510) at src/switch_core.c:221
#8  0x00007f22f5095b0b in switch_scheduler_execute (tp=tp@entry=0x14a0510) at src/switch_scheduler.c:64
#9  0x00007f22f5096141 in task_own_thread (thread=<optimized out>, obj=0x14a0510) at src/switch_scheduler.c:95
#10 0x00007f22f537798c in dummy_worker (opaque=0x7f22e80026e8) at threadproc/unix/thread.c:151
#11 0x00007f22f2533dd5 in start_thread () from /lib64/libpthread.so.0
#12 0x00007f22f225cead in clone () from /lib64/libc.so.6

**Comment by dragos-oancea** [comment]
is this on Debian ? if not, what distro ?

**Comment by andywolk** [comment]
Fixed in version 1.10.6 https://github.com/signalwire/freeswitch/commit/913551315b4331f9e3efb67e7c04b15a894cc53c

**Comment by andywolk** [comment]
Please try the latest and let me know if the problem exists still.
